src/common/dav/memops.c

```diff
@@ -128,7 +128,7 @@ operation_log_persistent_init(struct operation_log *log,
 	}
 
 	/* initialize underlying redo log structure */
-	src->capacity = ulog_base_nbytes;
+	src->capacity = ULOG_BASE_SIZE;
```

Comment:
This looks really confusing to me. In dav_create_clogs(), we call ulog_construct_new() with LANE_REDO_EXTERNL_SIZE, but here we specify it as ULOG_BASE_SIZE?

It looks to me all the 'lane' stuff is unused? should we remove these LANE_REDO/UNDO size macros and always use ULOG_BASE_SIZE for creating & extending?

Reply:
The lane stuff in pmdk is to persist the redo and undo logs. The redo ops are initially logged into a transient memory logs (pshadow_ops and transient_ops) and later persisted to lane pointed by ctx->ulog before committing the changes. The undo logs are written directly to ulog as and when it happens.

With BMEM, there is no need to persist these logs and redo ops in pshadow_ops are converted to wal redo ops and not written to ctx->ulog. Hence the ulog_construct_new() with LANE_REDO_EXTERNL_SIZE can be cleaned up. The cleanup requires some more changes as certain functions expects ctx->ulog to be not NULL. @ivadmitry is looking at doing the cleanup with changes to the functions that operates on ctx->ulog.